src/realm.h

```diff
@@ -1872,4 +1872,40 @@ RLM_API realm_notification_token_t* realm_results_add_notification_callback(real
  */
 RLM_API realm_results_t* realm_results_from_thread_safe_reference(const realm_t*, realm_thread_safe_reference_t*);
 
+/* Logging */
+typedef struct realm_logger realm_logger_t;
+
+typedef enum realm_log_level {
+    RLM_LOG_LEVEL_ALL,
+    RLM_LOG_LEVEL_TRACE,
+    RLM_LOG_LEVEL_DEBUG,
+    RLM_LOG_LEVEL_DETAIL,
+    RLM_LOG_LEVEL_INFO,
+    RLM_LOG_LEVEL_WARNING,
+    RLM_LOG_LEVEL_ERROR,
+    RLM_LOG_LEVEL_FATAL,
+    RLM_LOG_LEVEL_OFF,
```

Comment:
Shouldn't `OFF` be the first case?

Reply:
It follows the original `util::Logger::Level` declaration order and goes from most verbose (`all`) to least verbose (`off`).

Reply:
Hmm... I would have conceptualised it as the opposite 😅 . I would assume the more common use case would be first, and the most common use case to me would be to have logging off.

Reply:
Really, the common case is to not log even in the fatal case? That seems unfortunate.

But the main thing here is that they are defined this way so that `should_log(level)` is `level >= configured_threshold`. So more "important" levels need to have a higher number (with "off" being higher than any actual log line). Theoretically, we could reverse it all so that the test uses `<=` and "more important" levels have lower numbers. I doubt it is worth making that change though.

Reply:
Fair point on the fatal case– I was thinking about release builds where the users would (likely) blanketly turn logging off, but I suppose we'd actually want to encourage it being on for the fatal case. Either way, as you said, probably not worth making the change.